freqtrade/plot/plotting.py

```diff
@@ -358,7 +358,7 @@ def load_and_plot_trades(config: Dict[str, Any]):
         )
 
         store_plot_file(fig, filename=generate_plot_filename(pair, config['ticker_interval']),
-                        directory=config['user_data_dir'] / "plot")
```

Comment:
i would leave this as `/` - see other comment as to why `user_data_dir` is wrong in your environment.

Reply:
I can use `Path()` in the function call here too, but the concatenation seems sloppy. As a Windows user, I am scared of `/`.

Reply:
Please have a look at the [pathlib](https://docs.python.org/3/library/pathlib.html#basic-use) documentation.

It's main purpose is to make handling cross-compatibility easy by abstracting this away - using `Path("xxx") / "otherpath"` is the same as `Path("xxx").joinpath("otherpath")`.

This did never really work with `os.path` and could create problems - that's why i'm trying to move avay from using `os.path` (slowly, but i try to replace usage whenever i am working on something using `os.path`).

if you look at the implementation of the 2 methods (below) (`/` is implemented as `__truediv__()`) you'll notice that they call exactly the same method internally - so are factually identical.

https://github.com/python/cpython/blob/e0a1561f93545f38034d6c48f21e79acfd4db04b/Lib/pathlib.py#L899-L908

Reply:
Ah, ok. Thanks for that explanation. So, the opaque requirement is that `config['somepath']` is a path object and not a string. That conversion is bypassed when I set dictionary values directly in a notebook with `config['newpath'] = newpath`.

However, that doesn't explain the behavior in `freqtrade/resolvers/strategy_resolver.py ` noted below. `abs_path` should be pulling a path object from the config object since I'm not messing with the key:value pair after using `config_from_file()`. Perhaps `config['user_data_dir']` is not being converted to a path object in `config_from_file()`?

It seems like a solution could be to create a config editing function for overriding config values inline. For example:
```python
# generate config object
config = config_from_files(config_files)
# this should now be true
type(config['userdir']) == Path object

# override some value 
config['userdir'] = config_value_from_str('user_data/user_repo')

# or
config['strategy'] = config_value_from_str('NewStrategy')
```
Where `config_value_from_str` does the appropriate type interpolation and updates the config dict.

I wonder if that is the cause of the error noted in #2222 where `__slots__ = ['strategy']` gets confused about type?

Reply:
I think config_from_files does not do what you expect it to do - which is return a fully-initialized config object.

it returns merged dicts - nothing more, nothing less. That's partially intentional, since user_data_dir will create subdirectories if necessary ... which may not be desired for notebook usage.

I'll think of a solution to this outside of this PR - for now i would set `config['user_data_dir'] = Path("somepath")` to mitigate problems you're encountering - especially since you seem to (for now) be the only one with problems in this regard... (however we will fix this eventually, but a little (bigger?) change to config might be required for that to work correctly.

Reply:
That makes complete sense. As I have said, these practical challenges are about expanding my understanding of Python and how experienced developers solve common problems. I am more interested in the process than the solution. I look forward to seeing how you tackle this issue and am willing to help with whatever is useful to the project.